sympy/functions/elementary/integers.py

```diff
@@ -134,10 +134,9 @@ def _eval_number(cls, arg):
     def _eval_as_leading_term(self, x, cdir=0):
         arg = self.args[0].as_leading_term(x)
```

Comment:
It is better to call `as_leading_term` only when we know that `arg0` is infinite to save time.